tests/python/gaia-ui-tests/gaiatest/apps/system/regions/cards_view.py

```diff
@@ -65,19 +67,22 @@ def is_app_present(self, app):
         return self.is_element_present(*self._app_card_locator(app))
 
     def tap_app(self, app):
-        self.wait_for_condition(lambda m: self.is_app_displayed(app))
+        Wait(self.marionette).until(lambda m: self.is_app_displayed(app))
         self.marionette.find_element(*self._app_card_locator(app)).tap()
 
     def close_app(self, app):
         self.wait_for_card_ready(app)
         self.marionette.find_element(*self._close_button_locator(app)).tap()
-        self.wait_for_element_not_present(*self._app_card_locator(app))
+        Wait(self.marionette).until(
+            expected.element_not_present(*self._app_card_locator(app)))
```

Comment:
Locate the element before we close it, that way we verify it was visible and can wait for it to be not visible.